crypto/bio/bio_lib.c

```diff
@@ -714,12 +714,20 @@ long BIO_callback_ctrl(BIO *b, int cmd, BIO_info_cb *fp)
  */
 size_t BIO_ctrl_pending(BIO *bio)
 {
-    return BIO_ctrl(bio, BIO_CTRL_PENDING, 0, NULL);
+    long ret = BIO_ctrl(bio, BIO_CTRL_PENDING, 0, NULL);
+
+    if (ret < 0)
+        ret = 0;
+    return (size_t)ret;
```

Comment:
Is it always safe to assume that `LONG_MAX <= SIZE_MAX`? Should there be and additional overflow check here?

Reply:
> IMO we're asking for "condition is always true" or similar warnings from compilers or coverity with such checks

We can conditionally compile such checks if we want them, e.g. we have this at one point in the code:

````
#if LONG_MAX > INT_MAX
    /* Check that we did not truncate the length */
    if (len > INT_MAX) {
        ERR_raise(ERR_LIB_PEM, PEM_R_HEADER_TOO_LONG);
        return 0;
    }
#endif
````

> AFAIK our code depends on LONG_MAX <= SIZE_MAX.

That's news to me if so. Do you have an example of where we rely on that? I'm not entirely sure if its a reasonable assumption. The only sanity tests I can see that we have in this area are:

````
    /* This isn't possible to check using the framework functions */
    if (SIZE_MAX < INT_MAX) {
        TEST_error("int must not be wider than size_t");
        return 0;
    }

    /* SIZE_MAX is always greater than 2*INT_MAX */
    if (SIZE_MAX - INT_MAX <= INT_MAX) {
        TEST_error("SIZE_MAX must exceed 2*INT_MAX");
        return 0;
    }
````

Reply:
It doesn't seem that unreasonable to me to have a 32-bit system (which implies a 32-bit size_t), but that supports 64-bit longs. I'm not sure if any such platform actually exists though.

Reply:
Ok, you've persuaded me, although I really think there would be more breakage than this on such system. But I do not have concrete examples from the code.